### PR TITLE
Fix keda notebook testing

### LINKED ISSUE #2563: KEDA prom auto scale notebook is broken
## Describe the bug

The notebook testing keda_prom_auto_scale.ipynb is broken 

## To reproduce

make test_notebooks

## Expected behaviour

Pass test

## Environment

master branch

## Model Details 
* Images of your model: [Output of: `kubectl get seldondeployment -n <yourmodelnamespace> <seldondepname> -o yaml | grep image:` where `<yourmodelnamespace>`]
* Logs of your model: [You can get the logs of your model by running `kubectl logs -n <yourmodelnamespace> <seldonpodname> <container>`]

### PR DESCRIPTION
**Pull request by anggao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2563

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @anggao. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/SeldonIO/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **gsunner**
You can assign the PR to them by writing `/assign @gsunner` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by anggao** [comment]
@cliveseldon @RafalSkolasinski  FYI

**Comment by RafalSkolasinski** [comment]
Thanks @anggao!
/test notebooks

**Comment by seldondev** [comment]
Wed Oct 21 14:35:35 UTC 2020
The logs for [notebooks] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=1

**Comment by seldondev** [comment]
Wed Oct 21 14:35:58 UTC 2020
The logs for [integration] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=2

**Comment by seldondev** [comment]
@anggao: Cannot trigger testing until a trusted user reviews the PR and leaves an `/ok-to-test` message.

<details>

In response to [this](https://github.com/SeldonIO/seldon-core/pull/2565#issuecomment-713640133):

>/test lint


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by anggao** [comment]
@RafalSkolasinski can you add /ok-to-test for me :)

**Comment by seldondev** [comment]
Wed Oct 21 15:16:37 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=4

**Comment by seldondev** [comment]
Wed Oct 21 15:16:48 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=3

**Comment by seldondev** [comment]
@anggao: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test notebooks`
integration | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Wed Oct 21 17:02:40 UTC 2020
The logs for [integration] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=6

**Comment by seldondev** [comment]
Wed Oct 21 17:02:45 UTC 2020
The logs for [notebooks] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=5

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test integration`
notebooks | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test notebooks`
integration | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by anggao** [comment]
I found 
`test_keda_prom_auto_scale passed 1 out of the required 1 times. Success!`

Looks like the failure just because flaky tests ?

File: helm-charts/seldon-core-operator/templates/clusterrole_seldon-manager-role.yaml
Comment (cliveseldon) [inline helm-charts/seldon-core-operator/templates/clusterrole_seldon-manager-role.yaml:94]: Why did you remove v1alpha1 in the logs it seems to be saying
```
E1021 11:49:08.797354       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/reflector.go:125: Failed to list *v1alpha1.ScaledObject: scaledobjects.keda.sh is forbidden: User "system:serviceaccount:seldon-system:seldon-manager" cannot list resource "scaledobjects" in API group "keda.sh" at the cluster scope
```

**Review by cliveseldon — CHANGES_REQUESTED** [review]

**Comment by RafalSkolasinski** [comment]
> I found
> `test_keda_prom_auto_scale passed 1 out of the required 1 times. Success!`
> 
> Looks like the failure just because flaky tests ?

I am afraid not, the integration tests started failing and they were fine before :(

Currently we know about three other notebook tests failing:
- custom_metrics
- protocol_examples
- explainer_examples

everything should be green

File: helm-charts/seldon-core-operator/templates/clusterrole_seldon-manager-role.yaml
Comment (anggao) [reply helm-charts/seldon-core-operator/templates/clusterrole_seldon-manager-role.yaml:94]: I think in RBAC, apiGroups shouldn't include the version, e.g. https://github.com/SeldonIO/seldon-core/blob/master/helm-charts/seldon-core-operator/templates/clusterrole_seldon-manager-role.yaml#L102

https://github.com/kedacore/keda/blob/v2/config/rbac/role.yaml#L53

**Comment by anggao** [comment]
@RafalSkolasinski hum... I will take a look

**Comment by seldondev** [comment]
Thu Oct 22 10:58:07 UTC 2020
The logs for [integration] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=8

**Comment by seldondev** [comment]
Thu Oct 22 10:58:18 UTC 2020
The logs for [notebooks] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=7

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test integration`
notebooks | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test notebooks`
integration | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Thu Oct 22 13:37:02 UTC 2020
The logs for [integration] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=9

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Thu Oct 22 14:35:43 UTC 2020
The logs for [lint] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=11

**Comment by seldondev** [comment]
Thu Oct 22 14:35:46 UTC 2020
The logs for [pr-build] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=10

**Comment by seldondev** [comment]
Thu Oct 22 14:36:08 UTC 2020
The logs for [integration] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=12

**Comment by seldondev** [comment]
Thu Oct 22 14:36:46 UTC 2020
The logs for [notebooks] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=13

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test notebooks`
integration | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 143376c4ca7f0a5beefd962dfcfbecf2cc1e4190 | [link]() | `/test integration`
notebooks | 73ef93419a293c690866bc6b7cde0f380efe562b | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@anggao: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 73ef93419a293c690866bc6b7cde0f380efe562b | [link]() | `/test notebooks`
integration | 73ef93419a293c690866bc6b7cde0f380efe562b | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Thu Oct 22 16:41:54 UTC 2020
The logs for [lint] [15] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/15.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=15

**Comment by seldondev** [comment]
Thu Oct 22 16:41:58 UTC 2020
The logs for [pr-build] [14] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/14.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=14

**Comment by seldondev** [comment]
Thu Oct 22 16:45:33 UTC 2020
The logs for [integration] [16] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/16.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=16

**Comment by seldondev** [comment]
Thu Oct 22 16:45:43 UTC 2020
The logs for [notebooks] [17] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2565/17.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2565 --build=17

**Comment by seldondev** [comment]
@anggao: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 7c93b2afbd52e866c76dd0c701615df822c4704a | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by anggao** [comment]
Looks like only known notebook issue left: explainer_examples

**Comment by axsaucedo** [comment]
Awesome @anggao! This looks great - almost all green ticks! After this and Clive's prs are merged we'll be back to green ticks for all tests 🚀

**Pull request merged**
